example/ios/AdyenExampleTests/Modules/ApplePayModuleTests.swift

```diff
@@ -0,0 +1,208 @@
+  shippingMethod: .init(label: "Cargo", amount: 10, type: .final)
+)
+
+var mockPaymentRequest: PKPaymentRequest{
+  let paymentRequest = PKPaymentRequest()
+  paymentRequest.supportedNetworks = [.visa, .masterCard]
+  paymentRequest.countryCode = "US"
+  paymentRequest.currencyCode = "USD"
+  paymentRequest.merchantIdentifier = "merchant.com.test"
+  paymentRequest.paymentSummaryItems = [.init(label: "total", amount: 10)]
+  paymentRequest.merchantCapabilities = [.threeDSecure]
+  return paymentRequest
+}
+
+final class ApplePayModuleTests: XCTestCase {
+
+  var sut: ApplePayModule!
+  fileprivate var mockPaymentAuthorizationService: MockPKPaymentAuthorizationService!
+
+  override func setUp() {
+    super.setUp()
+    mockPaymentAuthorizationService = MockPKPaymentAuthorizationService()
+    sut = ApplePayModule(pkPaymentAuthorizationService: mockPaymentAuthorizationService)
+  }
+
+  override func tearDown() {
+    sut = nil
+    mockPaymentAuthorizationService = nil
+    super.tearDown()
+  }
+
+  func testIsAvailableSuccess() throws {
+      // Given
+    let sut = ApplePayModule(pkPaymentAuthorizationService: mockPaymentAuthorizationService)
+    mockPaymentAuthorizationService.authorizationViewControllerResult = PKPaymentAuthorizationViewController(paymentRequest: mockPaymentRequest)
+
+      // Then
+    sut.isAvailable(paymentMethodDict, configuration: configuration) { resolve in
+      let isSuccess = try! XCTUnwrap(resolve as? Bool )
+      XCTAssertTrue(isSuccess)
+    } rejecter: { title, message, error in
+      XCTFail("Shou not throw error")
+    }
+  }
+
+  func testIsAvailableReturnFalseWhenNoAmount() throws {
+      // Given
+    mockPaymentAuthorizationService.canMakePaymentsResult = false
+    let sut = ApplePayModule(pkPaymentAuthorizationService: mockPaymentAuthorizationService)
+    let configuration: NSDictionary = [
+      "clientKey": "test_client_key",
+      "countryCode": "US",
+      "applepay": [
+        "merchantID": "merchant.com.adyen.test",
+        "merchantName": "Test Merchant"
+      ]
+    ]
+
+      // Then
+    sut.isAvailable(paymentMethodDict, configuration: configuration) { resolve in
+      let isSuccess = try! XCTUnwrap(resolve as? Bool )
+      XCTAssertFalse(isSuccess)
+    } rejecter: { title, message, error in
+      XCTFail("Shou not throw error")
+    }
+  }
+
+  func testIsAvailableReturnFalseWhenNoViewController() throws {
+      // Given
+    mockPaymentAuthorizationService.canMakePaymentsResult = false
+    let sut = ApplePayModule(pkPaymentAuthorizationService: mockPaymentAuthorizationService)
+
+      // Then
+    sut.isAvailable(paymentMethodDict, configuration: configuration) { resolve in
+      let isSuccess = try! XCTUnwrap(resolve as? Bool )
+      XCTAssertFalse(isSuccess)
+    } rejecter: { title, message, error in
+      XCTFail("Shou not throw error")
+    }
+  }
+
+  func testApplePayDetailsExtraData() {
+      // When
+    let extraData = mockApplePayDetails.extraData
```

Comment:
I find it difficult to understand the test value here, is this a test for mock?

Reply:
it tests `extraData` helper - it is custom code on RN